### PR TITLE
modulegen: generate code-workspace with json marshal

### PR DESCRIPTION
**Pull request by mmorel-35**
## What does this PR do?

Replaces the template with the marshalling of a Config Object.
The path can be relative so it is relative to the .vs-code folder. As it is platform independant, it can be shared in the project.

## Why is it important?

This allows to define [multi-root-workspaces](https://code.visualstudio.com/docs/editor/multi-root-workspaces) in VSCode.

## Related issues

- Relates #1549

I tried it on windows plateform. @mdelapenya if you do have another platform, can you confirm, this works ?

### CODE REVIEW

**Review by mdelapenya — DISMISSED** [review]
Hi @mmorel-35 I went ahead and pushed a commit simply adding the new file to the module docs: https://github.com/testcontainers/testcontainers-go/pull/1551/commits/6e4d7f7d07c2e6f80b0da8037cd17b4103239192

I think this PR LGTM! Thanks again!

**Comment by mdelapenya** [comment]
I still need to get used to seeing the core module at the bottom, but I like the improvement. I guess you forced the commit of the workspace file, right?

**Comment by mmorel-35** [comment]
Probably something like that as the .gitignore doesn’t say anything about .vscode.
For the ordering, it could be changed to follow the path order if you prefer ?

File: modulegen/internal/vscode/types.go [hunk 1]
```diff
@@ -1,34 +1,43 @@
...
-		ModuleGeneratorDir: moduleGeneratorDirAbs,
-		Examples:           examples,
-		Modules:            modules,
+	for _, module := range modules {
+		config.Folders = append(config.Folders, Folder{
+			Path: "../modules/" + module,
+			Name: "modules/" + module,
+		})
 	}
+	sort.Slice(config.Folders, func(i, j int) bool { return config.Folders[i].Name < config.Folders[j].Name })
```
Comment (mmorel-35) [inline modulegen/internal/vscode/types.go:41]: You can replace Name by Path or define different names so they follow an order you like.

Comment (mdelapenya) [reply modulegen/internal/vscode/types.go:41]: In that case, let's have core library at the beginning. modulegen can be in its own place, but let's keep core with more visibility

Comment (mmorel-35) [reply modulegen/internal/vscode/types.go:41]: I'm  on it

File: .vscode/.testcontainers-go.code-workspace [hunk 2]
```diff
@@ -0,0 +1,113 @@
+// This file is autogenerated by the 'modulegen' tool.
+{
+    "folders": [
+        {
+            "name": "core",
```
Comment (mdelapenya) [inline .vscode/.testcontainers-go.code-workspace:5]:
```suggestion
            "name": "testcontainers-go",
```

File: modulegen/internal/vscode/types.go [hunk 3]
```diff
@@ -1,34 +1,43 @@
...
+}
 
-	for i, example := range examples {
-		examples[i] = filepath.Join(rootDirAbs, "examples", example)
+func newConfig(examples []string, modules []string) *Config {
+	config := Config{
+		Folders: []Folder{
+			{
+				Path: "../",
+				Name: "core",
```
Comment (mdelapenya) [inline modulegen/internal/vscode/types.go:21]:
```suggestion
				Name: "testcontainers-go",
```

**Review by mdelapenya — APPROVED** [review]
Super tiny change  in a name and we are done

File: modulegen/internal/vscode/types.go (hunk 3 above)
Comment (mdelapenya) [reply modulegen/internal/vscode/types.go:21]: We probably need to hack the sort process

Comment (mmorel-35) [reply modulegen/internal/vscode/types.go]: OK, It's done ;)

**Review by mdelapenya — APPROVED** [review]
Cool, thanks! LGTM

**Pull request merged**